airbyte_cdk/sources/declarative/concurrent_declarative_source.py

```diff
@@ -206,10 +210,34 @@ def _group_streams(
                 == "SimpleRetriever"
                 or name_to_stream_mapping[declarative_stream.name]["retriever"]["type"]
                 == "AsyncRetriever"
+                or name_to_stream_mapping[declarative_stream.name]["retriever"]["type"]
+                == "StateDelegatingRetriever"
             ):
                 incremental_sync_component_definition = name_to_stream_mapping[
                     declarative_stream.name
-                ].get("incremental_sync")
+                ].get("incremental_sync", {})
+
+                if (
+                    name_to_stream_mapping[declarative_stream.name]
+                    .get("retriever", {})
+                    .get("full_refresh_no_slice_in_params", False)
+                    and incremental_sync_component_definition
+                ):
+                    incremental_sync_component_definition["step"] = None
+                    incremental_sync_component_definition["cursor_granularity"] = None
+                    incremental_sync_component_definition["start_time_option"] = None
+                    incremental_sync_component_definition["end_time_option"] = None
+
+                if (
+                    name_to_stream_mapping[declarative_stream.name]
+                    .get("retriever", {})
+                    .get("full_refresh_ignore_min_max_datetime", False)
+                    and incremental_sync_component_definition
+                ):
+                    incremental_sync_component_definition["start_datetime"]["max_datetime"] = None
+                    incremental_sync_component_definition["start_datetime"]["min_datetime"] = None
+                    incremental_sync_component_definition["end_datetime"]["max_datetime"] = None
+                    incremental_sync_component_definition["end_datetime"]["min_datetime"] = None
 
                 partition_router_component_definition = (
```

Comment:
We have a very interesting case with @tolik0 where we would need to have a case with source-monday where the full refresh is just an incremental sync and the incremental has a partition router. Surprisingly, the fact that it would be different state format (full refresh is a simple `{<cursor_field>: <cursor_value>}` but the incremental one is per partition) would be fine because we only case about the parent state in this case and the migration from "simple" to "per partition" sets `<cursor_value>` to the parent state. However in order for that to work, we would need to check if there is a state and if so if there is a partition router on the incremental retriever. Do you think it would be possible?

Reply:
After reverting to the two-retriever instances approach, I think it could work.

Is this stream what you were describing?
**https://github.com/airbytehq/airbyte/blob/fe3ea86ec1508622b5e8e261586cb4b66b89eafd/airbyte-integrations/connectors/source-stripe/source_stripe/manifest.yaml#L702**